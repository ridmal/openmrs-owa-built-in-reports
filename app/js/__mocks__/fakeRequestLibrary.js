import Response from 'http-response-object';

export const fakeRequestLibrary = (requestUrl, requestOptions, shouldPass = true, responseData = null) => {

    if(shouldPass){
        return new Response(200, {}, responseData || { message: `You called ${requestUrl}` }, requestUrl);
    }else{
        return new Response(404, {}, responseData || { message: `The page at  ${requestUrl} was not found` }, requestUrl);
    }
};
