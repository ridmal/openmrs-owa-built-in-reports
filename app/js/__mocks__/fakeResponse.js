

export const FAKE_RESPONSE = {
    dataSets: [
        {
            metadata: {
                columns: [
                    {
                        name: "username",
                        label: "username"
                    },
                    {
                        name: "date_created",
                        label: "date_created"
                    },
                    {
                        name: "count",
                        label: "count"
                    }
                ]
            },
            rows: [
                {
                    date_created: "2010-04-26T13:25:00.000+0530",
                    username: "daemon",
                    count: 2
                },
                {
                    date_created: "2017-06-08T21:37:18.000+0530",
                    username: "clerk",
                    count: 13
                },
                {
                    date_created: "2017-07-08T21:37:18.000+0530",
                    username: "nurse",
                    count: 5
                },
                {
                    date_created: "2017-07-08T21:37:19.000+0530",
                    username: "doctor",
                    count: 1
                },
                {
                    date_created: "2017-07-08T21:37:18.000+0530",
                    username: "sysadmin",
                    count: 4
                }
            ]
        }
    ],
    definition: {
        name: 'test report'
    }

};