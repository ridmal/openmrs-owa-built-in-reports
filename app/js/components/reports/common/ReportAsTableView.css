@import url('https://fonts.googleapis.com/css?family=Open+Sans');


.react-grid-HeaderCell{
      background-color: #ff6666;
      color: white;
      font-family: 'Open Sans', sans-serif;
      text-align: center;
}

.react-grid-Cell{
      text-align: center;
}

.TableDivHolder {
      border-top: 1px solid;
      overflow: hidden;
}