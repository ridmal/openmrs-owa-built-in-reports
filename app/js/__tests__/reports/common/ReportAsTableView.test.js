import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import ReactDataGrid from 'react-data-grid';
import { fakeRequestLibrary } from '../../../__mocks__/fakeRequestLibrary';
import { FAKE_RESPONSE } from '../../../__mocks__/fakeResponse';
import ReportAsTableView from '../../../components/reports/common/ReportAsTableView';

jest.mock('react-data-grid', () => 'ReactDataGrid');

describe('<ReportAsTableView /> ', () => {
    it('component displays as expected', () => {
        const params = {
            retired: false
        };

        jest.mock('react-data-grid');
        var rendered = renderer.create(
            <ReportAsTableView reportUUID="e451ae04-4881-11e7-a919-92ebcb67fe33"
                reportParameters={params}
                fetchData={fakeRequestLibrary('www.openmrs-fake-server.org', {}, true, FAKE_RESPONSE)} />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});