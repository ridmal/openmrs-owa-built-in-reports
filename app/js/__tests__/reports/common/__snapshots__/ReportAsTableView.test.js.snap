// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportAsTableView />  component displays as expected 1`] = `
<div
  className="TableDivHolder"
>
  <ReactDataGrid
    columns={
      Array [
        Object {
          "key": "username",
          "name": "username",
          "resizable": true,
        },
        Object {
          "key": "date_created",
          "name": "date created",
          "resizable": true,
        },
        Object {
          "key": "count",
          "name": "count",
          "resizable": true,
        },
      ]
    }
    minHeight={280}
    rowGetter={[Function]}
    rowsCount={5}
  />
</div>
`;
